app/forms/schools/add_participant_form.rb

```diff
@@ -76,16 +83,34 @@ def email_already_taken?
     end
 
     def type=(value)
+      reset_steps(:details, :choose_mentor) if value.to_s != type
+
+      super(value&.to_sym)
+      if type == :self
+        self.full_name = current_user.full_name
+        self.email = current_user.email
+        self.participant_type = :mentor
+      else
+        self.participant_type = type
+      end
+    end
+
+    def participant_type=(value)
       super(value&.to_sym)
     end
 
     def school_cohort
       @school_cohort ||= SchoolCohort.find_by(id: school_cohort_id)
     end
 
+    def current_user
+      @current_user ||= User.find_by(id: current_user_id)
+    end
+
     def save!
       ActiveRecord::Base.transaction do
-        user = User.create!(full_name: full_name, email: email)
+        # TODO: What if email matches but with different name?
```

Comment:
Validation error?

Reply:
"This email does not match that user's name. Guess again!"